src/utils/miscellaneous.ts

```diff
@@ -14,7 +27,7 @@ export function getToken<T extends TokenDetails, K extends keyof T>(
   return (tokens || []).find((token: T): boolean => {
     const tokenKeyValue = token[key]
     if (tokenKeyValue) {
-      return typeof tokenKeyValue === 'string' && tokenKeyValue.toString().toUpperCase() === valueUppercase
+      return tokenKeyValue?.['toString']().toUpperCase() === valueUppercase
```

Comment:
Not `tokenKeyValue?.toString().toUpperCase()`?

Reply:
if it's `toString` pass it auto applies `toUpperCase` - i tested it but maybe u right

Reply:
ah i see ur comment, i tried the former but it broke all the things. This was the only way I could get it to work and my google foo energy was low

Reply:
but, but they are [the same](https://www.typescriptlang.org/play/?ssl=5&ssc=25&pln=1&pc=1#code/LAKFBcHsGsFMDsDSsCeA1AhgGwK6wPwB0UAyuAE4CW8A5gBQCUxkAqgA5uzkDCGAzrEagIMBMnTY8RANoByUhWo1ZAXUbN2nHv0EMgA)

weird

Reply:
yep i know
/shrug

Reply:
resolving even tho it sucks